static/js/settings_org.js

```diff
@@ -563,19 +572,33 @@ exports.build_page = function () {
     set_user_invite_restriction_dropdown();
     set_message_content_in_email_notifications_visiblity();
 
+    function get_auth_method_table_data() {
+        var new_auth_methods = {};
+        var auth_method_rows = $("#id_realm_authentication_methods").find('tr.method_row');
+        _.each(auth_method_rows, function (method_row) {
+            new_auth_methods[$(method_row).data('method')] = $(method_row).find('input').prop('checked');
+        });
+        return new_auth_methods;
+    }
+
     function check_property_changed(elem) {
         elem = $(elem);
         var property_name = exports.extract_property_name(elem);
         var changed_val;
         var current_val = get_property_value(property_name);
-
         if (typeof current_val === 'boolean') {
             changed_val = elem.prop('checked');
         } else if (typeof current_val === 'string') {
             changed_val = elem.val().trim();
         } else if (typeof current_val === 'number') {
             current_val = current_val.toString();
             changed_val = elem.val().trim();
+        } else if (typeof current_val === 'object') {
+            // Currently we only deal with realm_authentication_methods object
+            current_val = sort_object_by_key(current_val);
```

Comment:
Actually here we want to compare `current_val` and `changed_val` in `check_property_changed` function of the property `realm_authentication_methods`, which are objects, and I found here https://stackoverflow.com/a/1144249 that there is no easy way to do so. 
So I followed this approach,
```js
 JSON.stringify(obj1) === JSON.stringify(obj2) 
```
but before converting them to string we want the same order of keys, so I created this function `sort_object_by_key`, and a similar thing was already done in `populate_auth_methods` so first I extracted this logic into a separate function (which I thought made that also more readable) and in next commit used it here.